io-engine/src/bin/io-engine.rs

```diff
@@ -84,6 +93,12 @@ fn start_tokio_runtime(args: &MayastorCliArgs) {
         warn!("Nexus reset is disabled");
     }
 
+    // Whether the nexus channel should have readers/writers configured.
+    // This must be set true ONLY from tests.
+    if let Ok(v) = std::env::var("INIT_THRD_NX_CHAN_HANDLES") {
```

Comment:
Btw, is it not enough to do this on the init stage?

Reply:
> Btw, is it not enough to do this on the init stage?

It should be enough, yes. Earlier I was only setting from inside `start_tokio_runtime()` and it wasn't reflecting in tests, because test is seting up environment via MayastorTest object directly.

> Why do we need to set this again if we've set it on L75?

Removing the changes from this function. Don't we need to handle env variable and cli arg both? I can probably remove env variable dependency completely from `structopt` for this option.

Reply:
Ah nice if you can do this by only changing the `MayastorEnvironment` and not cli arg I think that's even better

Reply:
Need to modify `MayastorTest` constructor like this. This will become a blanket setting `true` for all tests though, less flexible(should be ok?). Changing the signature of `MayastorTest` to take `MayastorEnvironment` won't be desirable as part of this change I'd say.

```            
@@ -75,7 +75,9 @@ impl<'a> MayastorTest<'a> {
         let thdl = std::thread::Builder::new()
             .name("mayastor_master".into()) 
           .spawn(move || {
-                MayastorEnvironment::new(args).init();
+                let mut ms_env = MayastorEnvironment::new(args);
+                ms_env.set_io_all_thrd_nexus_channels(true);
+                ms_env.init();
```

Reply:
Seems quite verbose though, then I'd suggest keep as is in cli args, hidden.